sdk/inc/azure/core/az_platform.h

```diff
@@ -16,9 +16,18 @@
 #define _az_PLATFORM_H
 
 #include <azure/core/az_result.h>
-
+#ifndef __APPLE__
+#include <azure/platform/internal/az_platform_internal.h>
```

Comment:
Instead of this, can't we just have null implementations on Apple for now with an explanation that callback timers aren't supported?